judge/migrations/0106_submission_lock.py

```diff
@@ -0,0 +1,44 @@
+# Generated by Django 2.2.12 on 2020-05-13 20:58
+
+from django.db import migrations, models
+from django.utils import timezone
+
+
+def updatecontestsubmissions(apps, schema_editor):
+    Contest = apps.get_model('judge', 'Contest')
+    for row in Contest.objects.all():
```

Comment:
Similar thing as with the contest admin action, is there a reason a `Contest.objects.filter(...).update(...)` can't be used here?

Reply:
Only base field are available in migrations, so the `ended` property has to be manually computed.

Reply:
`ended` doesn't seem to be used anywhere here. Only `end_time` and `is_locked` are used, which are both base fields (unless I'm misunderstanding something).